frontend/app/routes/protected/person-case/name-info.tsx

```diff
@@ -0,0 +1,226 @@
+  'birth-certificate',
+  'citizenship-certificate',
+] as const;
+
+export const handle = {
+  i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
+} as const satisfies RouteHandle;
+
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  return {
+    documentTitle: t('protected:primary-identity-document.page-title'),
+    defaultFormValues: {
+      ...context.session.inPersonSINCase?.nameInfo,
+    },
+  };
+}
+
+export function meta({ data }: Route.MetaArgs) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: Route.ActionArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const lang = getLanguage(request);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  const formData = await request.formData();
+  const action = formData.get('action');
+  const docsRequired = formData.get('docs-required') === 'Yes';
+  const maxLength = 100;
+
+  switch (action) {
+    case 'back': {
+      throw i18nRedirect('routes/protected/index.tsx', request);
+    }
+
+    case 'next': {
+      const schema = v.object({
+        firstName: v.pipe(
+          v.string(t('protected:name-info.first-name-error.required-error')),
+          v.trim(),
+          v.nonEmpty(t('protected:name-info.first-name-error.required-error')),
+          v.maxLength(maxLength, t('protected:name-info.first-name-error.max-length-error')),
+          v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:name-info.first-name-error.format-error')),
+        ),
+        middleName: v.optional(
+          v.pipe(
+            v.string(t('protected:name-info.middle-name-error.required-error')),
+            v.trim(),
+            v.maxLength(maxLength, t('protected:name-info.middle-name-error.max-length-error')),
+            v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:name-info.middle-name-error.format-error')),
+          ),
+        ),
+        lastName: v.pipe(
+          v.string(t('protected:name-info.last-name-error.required-error')),
+          v.trim(),
+          v.nonEmpty(t('protected:name-info.last-name-error.required-error')),
+          v.maxLength(maxLength, t('protected:name-info.last-name-error.max-length-error')),
+          v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:name-info.last-name-error.format-error')),
+        ),
+        docTypes: v.pipe(
+          v.array(v.string(), t('protected:name-info.supporting-error.required-error')),
+          docsRequired
+            ? v.nonEmpty(t('protected:name-info.supporting-error.required-error'))
+            : v.empty(t('protected:name-info.supporting-error.invalid-error')),
+          v.checkItems((item) => VALID_DOC_TYPES.includes(item), t('protected:name-info.supporting-error.invalid-error')),
+        ),
+      }) satisfies v.GenericSchema<NameInfoSessionData>;
```

Comment:
I don't like this. A wall of code is hard to read and reason about. Sadly, I think that our current validation conventions requires this be the case. Requiring pre-translated error strings means that the `t(..)` function must be available when the schema is defined, which severely limits how and where we can init these schemas.

I think we should reconsider using pre-translated error messages in the schemas (in the name of enhanced readability).

Reply:
I don't think this should block the PR since it's the validation messages pattern. We can investigate for another solution on the side.